server/apps/microservices/src/processors/user-deletion.processor.ts

```diff
@@ -0,0 +1,34 @@
+import { assetUtils, userUtils } from '@app/common';
+import { AssetEntity } from '@app/database/entities/asset.entity';
+import { UserEntity } from '@app/database/entities/user.entity';
+import { QueueNameEnum, userDeletionProcessorName } from '@app/job';
+import { IUserDeletionJob } from '@app/job/interfaces/user-deletion.interface';
+import { Process, Processor } from '@nestjs/bull';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Job } from 'bull';
+import { Repository } from 'typeorm';
+
+@Processor(QueueNameEnum.USER_DELETION)
+export class UserDeletionProcessor {
+  constructor(
+    @InjectRepository(UserEntity)
+    private userRepository: Repository<UserEntity>,
+
+    @InjectRepository(AssetEntity)
+    private assetRepository: Repository<AssetEntity>,
+  ) {}
+
+  @Process(userDeletionProcessorName)
+  async processUserDeletion(job: Job<IUserDeletionJob>) {
+    const { user } = job.data;
+    const userAssets = await this.assetRepository.find({ where: { userId: user.id } });
+    // just for extra protection here
+    if (userUtils.isReadyForDeletion(user)) {
+      for (const asset of userAssets) {
+        assetUtils.deleteFiles(asset);
+        this.assetRepository.remove(asset);
+      }
```

Comment:
This is technically fine, but it does bring up some good questions if the user has added a lot of assets. I'm not sure what impact it has deleting one at a time vs bulk deleting say 10k photos, for example.

Would it make more sense to do something like `assetRepository.delete({ where: { userId }})` and potentially `fs.rmSync(dir, { recursive: true, force: true });`?

Maybe I'll ask this on discord.

Reply:
Yes, I agree should have done that, will update the PR with this approach

Reply:
Hey, we had a conversation in discord about this, you're good with this approach for now. I think we're planning on refactoring some of the asset delete logic in the near future to use queues, especially as we introduce trash can feature for assets.

Reply:
made the changes anyway